pages/Livesite/schedule.js

```diff
@@ -1,23 +1,493 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Page from '../../components/page';
-import { getHackathons } from '../../utility/firebase';
-import { LIVESITE_NAVBAR } from '../../constants';
-
-export default ({ hackathons }) => (
-  <Page
-    currentPath="Livesite"
-    hackathons={hackathons}
-    navbarItems={LIVESITE_NAVBAR}
-  >
-    <img
-      src="https://i.imgur.com/lEjmB7U.gif"
-      width="480"
-      height="362"
-      alt="panik"
-    />
-    <p>Schedule - This area under construction.</p>
-  </Page>
-);
+import FeatureFlag from '../../components/FeatureFlag';
+import {
+  formatDate,
+  getTimestamp,
+  getHackathons,
+  getActiveHackathon,
+  getLivesiteEvents,
+  addLivesiteEvent,
+  updateLivesiteEvent,
+  deleteLivesiteEvent,
+} from '../../utility/firebase';
+import Card, {
+  CardHeader,
+  CardTitle,
+  CardButtonContainer,
+  CardContent,
+} from '../../components/card';
+import Button from '../../components/button';
+import {
+  TableWrapper,
+  TableContent,
+  TableRow,
+  TableHeader,
+  TableData,
+  ActionsButtonContainer,
+  TableEmptyText,
+} from '../../components/table';
+import Modal, { Label, ModalContent, ModalField } from '../../components/modal';
+import {
+  COLOR,
+  EDIT,
+  VIEW,
+  NEW,
+  DELETE,
+  LIVESITE_NAVBAR,
+} from '../../constants';
+import { DateTimePicker } from '../../components/dateTimePicker';
+import { useAuth } from '../../utility/auth';
+
+export default function Events({ hackathons }) {
+  const [activeHackathon, setActiveHackathon] = useState('');
+  const [events, setEvents] = useState([]);
+  const [newEvent, setNewEvent] = useState({});
+  const [eventViewing, setEventViewing] = useState({});
+  const [eventEditing, setEventEditing] = useState({});
+  const [eventConfirm, setEventConfirm] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [alertMsg, setAlertMsg] = useState('');
+  const { email: user } = useAuth().user;
+
+  useEffect(() => {
+    (async () => {
+      setActiveHackathon(await getActiveHackathon);
+    })();
+  });
+
+  const fetchEvents = async () => {
+    if (activeHackathon) {
+      const eventsFetched = await getLivesiteEvents(activeHackathon);
+      if (Object.keys(eventsFetched).length > 0) {
+        setEvents(eventsFetched);
+      }
+    }
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    if (alertMsg.length > 0) {
+      alert(alertMsg);
+    }
+  }, [alertMsg]);
+
+  useEffect(() => {
+    fetchEvents();
+  }, [activeHackathon]);
+
+  const handleNew = async () => {
+    newEvent.lastModifiedBy = user;
+    newEvent.startTime = new Date(newEvent.startTime).toISOString();
+    newEvent.endTime = new Date(newEvent.endTime).toISOString();
+    const eventID = await addLivesiteEvent(activeHackathon, { ...newEvent });
+    newEvent.lastModified = formatDate(getTimestamp().seconds);
+    setEvents({
+      ...events,
+      [eventID]: {
+        ...newEvent,
```

Comment:
I think `startTime` and `endTime` should be formatted here for consistency with the other pages. You probably saw this in events - I basically did `date: formatDate(eventEditing.date, true),`

I think the same should be applied to `handleUpdate`